Userland/Libraries/LibWeb/Fetch/Response.cpp

```diff
@@ -0,0 +1,309 @@
+{
+    auto& realm = window.realm();
+
+    // Referred to as 'this' in the spec.
+    auto response_object = [&] {
+        auto response = adopt_own(*new Infrastructure::Response());
+        return JS::NonnullGCPtr { *realm.heap().allocate<Response>(realm, realm, move(response)) };
+    }();
+
+    // 1. Set this’s response to a new response.
+    // NOTE: This is done at the beginning as the 'this' value Response object
+    //       cannot exist with a null Infrastructure::Response.
+
+    // 2. Set this’s headers to a new Headers object with this’s relevant Realm, whose header list is this’s response’s header list and guard is "response".
+    response_object->m_headers = realm.heap().allocate<Headers>(realm, window);
+    response_object->m_headers->set_header_list(response_object->response().header_list());
+    response_object->m_headers->set_guard(Headers::Guard::Response);
+
+    // 3. Let bodyWithType be null.
+    Optional<Infrastructure::BodyWithType> body_with_type;
+
+    // 4. If body is non-null, then set bodyWithType to the result of extracting body.
+    if (body.has_value())
+        body_with_type = TRY(extract_body(realm, *body));
+
+    // 5. Perform initialize a response given this, init, and bodyWithType.
+    TRY(response_object->initialize_response(init, body_with_type));
+
+    return response_object;
+}
+
+// https://fetch.spec.whatwg.org/#dom-response-error
+JS::NonnullGCPtr<Response> Response::error()
+{
+    auto& vm = Bindings::main_thread_vm();
+
+    // The static error() method steps are to return the result of creating a Response object, given a new network error, "immutable", and this’s relevant Realm.
+    // FIXME: How can we reliably get 'this', i.e. the object the function was called on, in IDL-defined functions?
+    return Response::create(Infrastructure::Response::network_error(), Headers::Guard::Immutable, *vm.current_realm());
+}
+
+// https://fetch.spec.whatwg.org/#dom-response-redirect
+WebIDL::ExceptionOr<JS::NonnullGCPtr<Response>> Response::redirect(String const& url, u16 status)
+{
+    auto& vm = Bindings::main_thread_vm();
+    auto& realm = *vm.current_realm();
+    auto& window = verify_cast<HTML::Window>(realm.global_object());
+
+    // 1. Let parsedURL be the result of parsing url with current settings object’s API base URL.
+    auto api_base_url = HTML::current_settings_object().api_base_url();
+    auto parsed_url = URLParser::parse(url, &api_base_url);
+
+    // 2. If parsedURL is failure, then throw a TypeError.
+    if (!parsed_url.is_valid())
+        return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, "Redirect URL is not valid"sv };
+
+    // 3. If status is not a redirect status, then throw a RangeError.
+    if (!Infrastructure::is_redirect_status(status))
+        return WebIDL::SimpleException { WebIDL::SimpleExceptionType::RangeError, "Status must be in range 200-299"sv };
```

Comment:
The Range Error text `Status must be in range 200-299` does not compute for me with this supposed to being a redirect status

Reply:
Another copy/paste mistake, it seems